### PR TITLE
using authentication keys instead of hardcoded email in 2fa

### PR DESCRIPTION
**Pull request by mauisu**
[#136768599]

### CODE REVIEW

**Comment by werleo** [comment]
@edbo are you ok with this change?
It is blocking agents feature.

**Comment by edbo** [comment]
I'm actually not super keen tying this implementation to our user model. Email kind of stinks too!

Ideally this would something really generic in this library and could be configured to something really specific in a config file.

**Comment by mauisu** [comment]
I dont think im quite following, what do you mean @edbo, that you would like to have the fields by which a user can log in, in a config file which is used here?

`authentication_keys` option is a feature which they added in some version (and it is mentioned in the readmes and FAQ on how to change the login field for a devise user). Unfortunately the `devise_google_authentication` was not updated such that it uses the new features, which this pr enables.

**Comment by werleo** [comment]
can we update this gem to latest version and keep or changes in the same time?

**Comment by mauisu** [comment]
https://github.com/AsteriskLabs/devise_google_authenticator/compare/master...bitgaming:master there seems to be a couple of bug fixes which shouldnt brake anything

**Comment by edbo** [comment]
Ideally can we fully revert back to their gem?

I just think we've hacked the crap out of this gem to make it work for our benefit and would be good if we didn't at all.

**Comment by mauisu** [comment]
I think we could do that
I will play with it today and get back to you

**Comment by edbo** [comment]
@mauisu Is this stuck now?

**Comment by mauisu** [comment]
didn't look into it yet as we had other tasks prioritized

**Comment by edbo** [comment]
@mauisu Apparently this is blocking logging in for agents, suggest we start to give this some priority.

**Comment by mauisu** [comment]
already in progress @edbo

File: lib/devise_google_authenticatable/models/google_authenticatable.rb [hunk 1]
```diff
@@ -52,10 +52,10 @@ def validate_token(token)
         def gauth_enabled?
           # Active_record seems to handle determining the status better this way
           if self.gauth_enabled.respond_to?("to_i")
-            if self.gauth_enabled.to_i == 0 || self.gauth_enabled == 'f'
```
Comment (werleo) [inline lib/devise_google_authenticatable/models/google_authenticatable.rb]: please make sure that admins, affiliates and player does not have `'f'` in gauth_enabled

Comment (mauisu) [inline lib/devise_google_authenticatable/models/google_authenticatable.rb]:
```
[1] gaming_admin »  'f'.to_i
=> 0
```
(as well as any other string which does not begin with a valid number)
I included this information in the commit message :)

**Comment by mauisu** [comment]
this is a comparison to master https://github.com/AsteriskLabs/devise_google_authenticator/compare/master...bitgaming:feature/pt136768599-agents_fixes
what I left out is:
1) configurable return path
2) authentication key rather then email cookie signing (it falls back to email because it is the default)

**Pull request merged**